sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -345,6 +417,39 @@ internal static IReadOnlyList<EntityRecognitionTasksItem> ConvertToEntityRecogni
             return collection;
         }
 
+        internal static ExtractKeyPhrasesActionResults ConvertToExtractKeyPhrasesActionResults(AnalyzeJobState jobState, IDictionary<string, int> idToIndexMap)
+        {
+            var collection = new List<ExtractKeyPhrasesResultCollection>();
+            foreach (KeyPhraseExtractionTasksItem task in jobState.Tasks.KeyPhraseExtractionTasks)
+            {
+                collection.Add(ConvertToExtractKeyPhrasesResultCollection(task.ResultsInternal, idToIndexMap));
+            }
+
+            return new ExtractKeyPhrasesActionResults(collection, jobState.LastUpdateDateTime, jobState.Errors, jobState.Errors.Any());
```

Comment:
jobState.Errors should be parsed to errors specific to this task

Reply:
same applies for the other `ConverToxxActionResults` methods

Reply:
I am not sure how this will be parsed, lets have this discussed as other languages need to do this as well.

Reply:
@mssfang  what is Java doing?
I am thinking on something similar to OM or healthcare entity relations where we parse the json pointer (represented in the `target` to the task in the specified index.
```
{
    "jobId": "d40dff79-65bd-455a-98b7-7ec25c0548f2_637479072000000000",
    "lastUpdateDateTime": "2021-02-03T23:28:50Z",
    "createdDateTime": "2021-02-03T23:28:47Z",
    "expirationDateTime": "2021-02-04T23:28:47Z",
    "status": "failed",
    "errors": [
        {
            "code": "InvalidRequest",
            "message": "Job task parameter value 2020-04-01 is not supported for model-version parameter for job task type KeyPhraseExtraction. Supported values latest,2020-07-01.",
            "target": "#/tasks/keyPhraseExtractionTasks/0"
        }
    ],
    "tasks": {
        "details": {
            "lastUpdateDateTime": "2021-02-03T23:28:50Z"
        },
        "completed": 0,
        "failed": 1,
        "inProgress": 0,
        "total": 1,
        "keyPhraseExtractionTasks": [
            {
                "lastUpdateDateTime": "2021-02-03T23:28:50.7809386Z",
                "results": {
                    "documents": [],
                    "errors": [],
                    "modelVersion": ""
                }
            }
        ]
    }
}
```

Reply:
`"target": "#/tasks/keyPhraseExtractionTasks/0"` has the task ID = 0 and we have the order of `keyPhraseExtractionTasks`. We can copy the error message and error code to TextAnalyticsActionResult as task error. I am also working on implementing the detail. I think I already show the idea to @suhas92 and I will keep it update once I have the implementation tomorrow.

Reply:
The java implementation will be like below, I have tested it out and it works fine. Working on creating a PR.

```
private static final String REGEX_ACTION_ERROR_TARGET = "#/tasks/(\\w+)/(\\d+)";
private String[] parseTarget(String targetReference) {
        if (CoreUtils.isNullOrEmpty(targetReference)) {
            throw logger.logExceptionAsError(new RuntimeException(
                "Expected an error with a target field referencing an action but did not get one"));
        }
        // action could be failed and the target reference is "#/tasks/keyPhraseExtractionTasks/0";
        final Pattern pattern = Pattern.compile(REGEX_ACTION_ERROR_TARGET, Pattern.MULTILINE);
        final Matcher matcher = pattern.matcher(targetReference);
        String[] pair = new String[2];
        while (matcher.find()) {
            pair[0] = matcher.group(1);
            pair[1] = matcher.group(2);
        }
        return pair;
    }
    
 if (!CoreUtils.isNullOrEmpty(keyPhraseExtractionTasks)) {
            for (int i = 0; i < keyPhraseExtractionTasks.size(); i++) {
                final TasksStateTasksKeyPhraseExtractionTasksItem taskItem = keyPhraseExtractionTasks.get(i);
                final ExtractKeyPhrasesActionResult actionResult = new ExtractKeyPhrasesActionResult();
                ExtractKeyPhrasesActionResultPropertiesHelper.setResult(actionResult,
                    toExtractKeyPhrasesResultCollection(taskItem.getResults()));
                TextAnalyticsActionResultPropertiesHelper.setCompletedAt(actionResult,
                    taskItem.getLastUpdateDateTime());
                extractKeyPhrasesActionResults.add(actionResult);
            }
        }

        final List<TextAnalyticsError> errors = analyzeJobState.getErrors();
        if (!CoreUtils.isNullOrEmpty(errors)) {
            for (TextAnalyticsError error : errors) {
                final String[] targetPair = parseTarget(error.getTarget());
                final String taskName = targetPair[0];
                final Integer taskIndex = Integer.valueOf(targetPair[1]);
                final TextAnalyticsActionResult actionResult;
                if ("entityRecognitionTasks".equals(taskName)) {
                    actionResult = recognizeEntitiesActionResults.get(taskIndex);
                } else if ("entityRecognitionPiiTasks".equals(taskName)) {
                    actionResult = recognizePiiEntitiesActionResults.get(taskIndex);
                } else if ("keyPhraseExtractionTasks".equals(taskName)) {
                    actionResult = extractKeyPhrasesActionResults.get(taskIndex);
                } else {
                    throw logger.logExceptionAsError(new RuntimeException(
                        "Invalid task name in target reference, " + taskName));
                }

                TextAnalyticsActionResultPropertiesHelper.setIsError(actionResult, true);
                TextAnalyticsActionResultPropertiesHelper.setError(actionResult,
                    new com.azure.ai.textanalytics.models.TextAnalyticsError(
                        TextAnalyticsErrorCode.fromString(
                            error.getCode() == null ? null : error.getCode().toString()),
                        error.getMessage(), null));
            }
        }
```

Reply:
thanks @mssfang ! appreciate it.